internal/action/setup.go

```diff
@@ -0,0 +1,287 @@
+
+	crypto := s.getCryptoFor(ctx, team)
+	if crypto == nil {
+		return fmt.Errorf("can not continue without crypto")
+	}
+	debug.Log("Crypto Backend initialized as: %s", crypto.Name())
+
+	// check for existing GPG keypairs (private/secret keys). We need at least
+	// one useable key pair. If none exists try to create one
+	if !s.initHasUseablePrivateKeys(ctx, crypto) {
+		out.Print(ctx, "🔐 No useable cryptographic keys. Generating new key pair")
+		out.Print(ctx, "🕰 Key generation may take up to a few minutes")
+		if err := s.initGenerateIdentity(ctx, crypto, ctxutil.GetUsername(ctx), ctxutil.GetEmail(ctx)); err != nil {
+			return errors.Wrapf(err, "failed to create new private key")
+		}
+		out.Print(ctx, "🔐 Cryptographic keys validated")
+	}
+
+	debug.Log("Has useable private keys")
+
+	// if a git remote and a team name are given attempt unattended team setup
+	if remote != "" && team != "" {
+		if create {
+			return s.initCreateTeam(ctx, team, remote)
+		}
+		return s.initJoinTeam(ctx, team, remote)
+	}
+
+	// assume local setup by default, remotes can be added easily later
+	return s.initLocal(ctx)
+}
+
+func (s *Action) initGenerateIdentity(ctx context.Context, crypto backend.Crypto, name, email string) error {
+	out.Green(ctx, "🧪 Creating cryptographic key pair (%s) ...", crypto.Name())
+
+	out.Print(ctx, "🎩 Gathering information for the key pair ...")
+	name, err := termio.AskForString(ctx, "🚶 What is your name?", name)
+	if err != nil {
+		return err
+	}
+
+	email, err = termio.AskForString(ctx, "📧 What is your email?", email)
+	if err != nil {
+		return err
+	}
+
+	passphrase := xkcdgen.Random()
+	pwGenerated := true
+	if bv, err := termio.AskForBool(ctx, "⚠ Do you want to enter a passphrase? (otherwise we generate one for you)", false); err != nil && bv {
+		pwGenerated = false
+		sv, err := termio.AskForPassword(ctx, "✍ Please enter your passphrase")
+		if err != nil {
+			return errors.Wrapf(err, "Failed to read passphrase")
+		}
+		passphrase = sv
+	}
+
+	out.Yellow(ctx, "⏳ This can take a long time. If you get impatient see https://github.com/gopasspw/gopass/blob/master/docs/entropy.md")
+	if want, err := termio.AskForBool(ctx, "Continue?", true); err != nil || !want {
+		return errors.Wrapf(err, "User aborted")
+	}
+
+	if err := crypto.GenerateIdentity(ctx, name, email, passphrase); err != nil {
+		return errors.Wrapf(err, "failed to create new private key in batch mode")
+	}
+
+	out.Print(ctx, "✅ Key pair generated")
+
+	if pwGenerated {
+		out.Print(ctx, color.MagentaString("Passphrase: ")+color.HiGreenString(passphrase))
+	}
+
+	// avoid the gpg cache or we won't find the newly created key
+	kl, err := crypto.ListIdentities(gpg.WithUseCache(ctx, false))
+	if err != nil {
+		return errors.Wrapf(err, "failed to list private keys")
+	}
+	if len(kl) > 1 {
+		out.Print(ctx, "⚠ More than one private key detected. Make sure to use the correct one!")
+		return nil
+	}
+	if len(kl) < 1 {
+		debug.Log("Private Keys: %+v", kl)
```

Comment:
We should never output private keys in logs, if you ask me... But in this case I'm not sure what that would output anyway since `len(kl)<0`

Reply:
No, we must never log secret material, but these lists should only contain key IDs.
However in this case it must always be empty, so removing it.